lib/inject.ex

```diff
@@ -8,19 +8,25 @@ defmodule Inject do
     :ok
   end
 
+  defmacro inject_module(mod) do
+    if Application.get_env(:inject, :disabled) do
+      mod
+    else
+      quote bind_quoted: [mod: mod] do
+        Inject.Registry
+        |> Registry.lookup(mod)
+        |> Enum.reverse()
+        |> find_override() || mod
+      end
+    end
+  end
+
   def i(mod) do
     inject(mod)
   end
 
   def inject(mod) do
-    if Application.get_env(:inject, :disabled) do
-      mod
-    else
-      Inject.Registry
-      |> Registry.lookup(mod)
-      |> Enum.reverse()
-      |> find_override() || mod
-    end
+    inject_module(mod)
```

Comment:
Almost there :)

Calling `inject_module` macro here would run the macro once during compilation and that's it. What we want is:

- An `inject_module` function (not macro) that does the injection (always, unconditionally)
- An `inject` macro, that based on application env config either does nothing or calls that `inject_module` function.